lib/chef/client.rb

```diff
@@ -415,6 +417,13 @@ def register_reporters
       end
     end
 
+    # Register the data collector reporter to send event information to the
+    # data collector server
+    # @api private
```

Comment:
Can you actually put this in the `private` section rather than making it `@api private`? I think the existing APIs we kept public in case someone was using, but we don't want the new APIs to be used.